src/modules/library/containers/Import.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+	CardBody, Row, Col,
+	Button, Input, Label,
+	FormGroup
+} from 'reactstrap';
+
+const Import = ({
+	api, app, setChosenPanel, retrieveAll
+}) => {
+	const { t } = useTranslation();
+	const [chosenFilename, setChosenFilename] = useState("");
+	const [type, setType] = useState("override");
+	const [errors, setErrors] = useState("");
+	const inputFileRef = useRef(null)
+	const formRef = useRef(null);
+
+	const chooseFile = () => {
+		if (!inputFileRef.current) return;
+
+		inputFileRef.current.click();
+	}
+
+	const updateFilename = () => {
+		if (!inputFileRef.current) return;
+
+		// Get filename from input and remove path
+		const filename = inputFileRef.current.value.replace(/.*[\/\\]/, '');
+		setChosenFilename(filename);
+
+		// Check if file is tar
+		if (!filename.includes(".tar")) {
+			setErrors(true);
+		} else {
+			setErrors(false);
+		}
+	}
+
+	const onTypeChange = (e) => setType(e.target.value);
+
+	const importLibrary = async event => {
+		event.preventDefault();
+		try {
+			const data = new FormData(formRef.current);
+			const response = await api.put(`/library/upload?type=${type}`, data, {
+				headers: {
+					'Content-Type': 'multipart/form-data'
+				}
+			})
+
+			if (response.data.result !== "OK") throw new Error(`Response result is ${response.data.result}. Expected result is OK`);
```

Comment:
@titkov49 Instead of `Expected result is OK` would be better to write that `File has not been uploaded`